models/productModel.js

```diff
@@ -23,6 +23,10 @@ const createProduct = async (data) => {
       data.designer_id, // not entirely sure if we'll be passing this value from our front-end
     ],
   );
+  await db.pool.asyncQuery(
+    'INSERT INTO Products_Categories (product_id, category_id) VALUES ((SELECT LAST_INSERT_ID()), ?)',
+    [data.category_id],
+  );
```

Comment:
@dlimx I'm not 100% if this is an acceptable implementation. Does having multiple `await` functions run them serially? Obvious we don't want the `INSERT` into Products_Categories to run until the `INSERT` into Products finishes.

Or is the preferred approach to have a `next` chained to the `createProduct()` call in the productService.js which would call `createProductCategory()` ...which we could define in productModel.js

Reply:
`async`/`await` will wait for each call to complete - so you can treat them as synchronous code

If we want to enforce that `category_id` should be always present, we can also do it within the same SQL call

Reply:
> `async`/`await` will wait for each call to complete - so you can treat them as synchronous code
Cool, so the code should work as is to INSERT a product/category relationship right after

> If we want to enforce that `category_id` should be always present, we can also do it within the same SQL call
Is `category_id` already required since we have that attribute set as NOT NULL?

Reply:
Yep, although other things could insert into Products/Categories in between

If it’s already required then it’s basically good to go

Reply:
> Yep, although other things could insert into Products/Categories in between

I think I understand what you mean. While it would be inconsequential, in code elsewhere in this app, there could be another INSERT statement being run on the Products_Categories around the same time that inserts a relationship before ours.